subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/recomp/GroovyRecompilationSpecProvider.java

```diff
@@ -41,11 +42,11 @@ public GroovyRecompilationSpecProvider(FileOperations fileOperations,
                                            FileTree sources,
                                            InputChanges inputChanges,
                                            Iterable<FileChange> sourceChanges,
-                                           GroovySourceFileClassNameConverter sourceFileClassNameConverter) {
+                                           Multimap<String, String> sourceClassesMapping) {
         super(fileOperations, sources);
         this.inputChanges = inputChanges;
         this.sourceChanges = sourceChanges;
-        this.sourceFileClassNameConverter = sourceFileClassNameConverter;
+        this.sourceFileClassNameConverter = new GroovySourceFileClassNameConverter(sourceClassesMapping);
```

Comment:
Shall we continue to pass in the `sourceFileClassNameConverter`? I think that would be better.

Reply:
Yeah... done.